File: sort/mergeSort/mergeSort.py
unSortedData = [10, 1, 9, 2, 8, 3, 7, 4, 6, 5, 0]

def mergeSort(data):
    # 基准条件
    if(len(data) <= 1):
        return data
    #递归条件 拆分成两个数组


    size = len(data)//2
    left = mergeSort(data[size:])
    right = mergeSort(data[:size])

    sortedData = []
    while( (len(left) > 0) and (len(right)>0)):
        if(left[0] <= right[0]):
            sortedData.append(left[0])
            del left[0]
        else:
            sortedData.append(right[0])
            del right[0]
    if(len(left) > 0):
        sortedData += left
    if(len(right) > 0):
        sortedData += right

    return sortedData


sortedData = mergeSort(unSortedData)
print(sortedData)

File: sort/mergeSort/mergeSort.go
package main

import "fmt"

func mergeSort(arr []int) []int {

	// 基线条件
	if len(arr) <= 1 {
		return arr
	}
	// 切分成两份

	var middleIndex = len(arr) / 2
	leftData := mergeSort(arr[middleIndex:])
	rightData := mergeSort(arr[:middleIndex])
	// 递归条件
	var sortedData []int
	for (len(leftData) > 0 && len(rightData) > 0) {
		if (leftData[0] <= rightData[0]) {
			sortedData = append(sortedData, leftData[0])
			leftData = leftData[1:]
		} else {
			sortedData = append(sortedData, rightData[0])
			rightData = rightData[1:]
		}
	}

	if (len(leftData) > 0) {
		sortedData = append(sortedData, leftData...)
	}
	if (len(rightData) > 0) {
		sortedData = append(sortedData, rightData...)
	}

	return sortedData
}

func main() {
	var unSortedData = []int{10, 1, 9, 2, 8, 3, 7, 4, 6, 5}
	var sortedData = mergeSort(unSortedData)
	fmt.Println(sortedData)
}
